internal/grpc/services/publicshareprovider/publicshareprovider.go

```diff
@@ -166,6 +167,7 @@ func (s *service) CreatePublicShare(ctx context.Context, req *link.CreatePublicS
 	share, err := s.sm.CreatePublicShare(ctx, u, req.ResourceInfo, req.Grant)
 	if err != nil {
 		log.Debug().Err(err).Str("createShare", "shares").Msg("error connecting to storage provider")
+		return nil, err
```

Comment:
Hm, this looks fishy. Shouldn't the error be returned as a non-OK status (+Message) in a proper CreatePublicShareResponse, here? (See e.g. link 186)
 
Also the log message seems at least missleading. How can we be sure that this is a connection issue?